doc/odoc_for_authors.mld

```diff
@@ -32,36 +32,36 @@ v}
 - {{:https://github.com/ocaml-doc/voodoo}Voodoo} is the driver used to create
   the docs for {{:https://ocaml.org/packages}the OCaml website}.
 
-{1:interfaces Documenting your interfaces}
+{1:interfaces Documenting Your Interfaces}
 
-Odoc is built to produce documentation for your {e libraries}, and the unit of
+[odoc] is built to produce documentation for your {e libraries}, and the unit of
 organisation is the {e module}. Documentation is written by putting
-{{!special_comments}special comments} into the source of the {e module interface} or
-less commonly your {e module implementation}. 
+{{!special_comments}special comments} into the source of the {e module interface} or,
+less commonly, your {e module implementation}. 
 
 For the HTML output, [odoc] will produce one page for each module, module type,
-class and class type. This includes any submodules, so if a module [A] contains
-module [B] then two HTML pages will be created, and [A/index.html] will have a
+class, and class type. This includes any submodules, so if a module [A] contains
+module [B] then two HTML pages will be created. [A/index.html] will have a
 link to [A/B/index.html] at the point that module [B] is defined. So each
-module, module type, class and class type should be documented keeping in mind
+module, module type, class, and class type should be documented, keeping in mind
 that it will end up as a single page with a table of contents. 
 
 For each module, [odoc] will describe all of the values, types, module types,
-classes, modules and all other elements found in module signatures, documenting
+classes, modules, and all other elements found in module signatures, documenting
 the details relevant to the OCaml type system along with any documentation
 written in specially formatted comments in the source code. For any element that
 references any other, for example a [value] that has a [type], [odoc] will
 create links such that clicking them will link to the definitions.
```

Comment:
```suggestion
classes, modules, and all other elements found in module signatures. These record
the details relevant to the OCaml type system along with any documentation
written in specially-formatted comments in the source code. For any element that
references any other, for example a [value] that has a [type], [odoc] will
create links such that clicking them will link to the definitions.
```

These are both long sentences. Could we possibly turn them into 3 sentences for readability like this?

Reply:
I think "documenting" was saying that these elements are added to the generated documentation, not just recorded. 
Also, "The details relevant to the OCaml type system" seems too abstract for no good reason.
I'd suggest something like that:

"The types or descriptions for these elements are documented along with any documentation comments present in the source code."

Reply:
Would "Their types or descriptions are added to the documentation comments in the source code" work?

Is it clear that "their" refers to the elements mentioned in the previous sentence?

Reply:
I think it's confusing to say that the "types or descriptions" are added to the "comments", the documentation is made from the "types or descriptions" and the "comments" at the same level (not one into the other).
The "Their" is fine.

Reply:
Okay! How about: 
"Their types or descriptions are documented along with any comments in the source code."
This suggests the comments are documented, too.

Does this say the same thing without using Document(ation) twice? Or should we stick with: "Their types or descriptions are documented along with any documentation comments present in the source code."

Reply:
I think it's fine to say "comments" instead of "documentation comments" but it adds a slight ambiguity as there's two types of comments (documentation and non-documentation).

Reply:
Oh! Interesting! Do you mean the difference between `**` (documentation) and `*` (non-documentation), @Julow?

Reply:
Yes that's correct. Comments that begin with a double asterisk - `(** ... *)` - are intended to end up in the published API documentation. Those that start with a single asterisk - `(* ... *)` are ignored by odoc.

Perhaps we could write "Their types or descriptions are documented along with any explicit documentation comments present in the source code."

Reply:
Would it be more clear to put in the `(** ... *)` as a reminder those are the documentation comments? 

"Their types or descriptions are documented along with any `(** ... *)` comments present in the source code.

(I'm looking for a way to avoid saying document(ation) twice in the same sentence, but if none of these work, then we'll stick with both. :-)

Reply:
Let's resolve this and merge this PR! We have a few propositions:

> Their types or descriptions are documented along with any docstring found in the source code.

or

> Their types or descriptions are documented along with any (** ... *) comments found in the source code.

or

> Their types or descriptions are documented along with any explicit documentation comments present in the source code.

or

> The types or descriptions for these elements are documented along with any documentation comments present in the source code.

I'm fine with all of them.

Reply:
> The "documentation comments" are often called "docstrings". Would it be OK to use that to avoid the "documentation" repetition?
> 
> > "Their types or descriptions are documented along with any docstring found in the source code."

I think docstrings really makes it more clear, as long as the reader knows that `docstring` = `(** )` comments